### PR TITLE
Update docs & fix bug in constraint

### PR DESCRIPTION
**Pull request by andreeaflorescu**
### Summary of the PR

This PR is the first in a series for publishing the crate on crates.io.
More details about the changes can be found in the commit messages.

### Requirements

Before submitting your PR, please make sure you addressed the following
requirements:

- [x] All commits in this PR are signed (with `git commit -s`), and the commit
  message has max 60 characters for the summary and max 75 characters for each
  description line.
- [x] All added/changed functionality has a corresponding unit/integration
  test.
- [x] Any newly added `unsafe` code is properly documented.

### CODE REVIEW

**Review by gsserge — APPROVED** [review]

File: src/address_allocator.rs [hunk 1]
```diff
@@ -47,22 +47,22 @@ impl AddressAllocator {
     }
 
     /// Allocates a new aligned memory slot. Returns the allocated range in case of success.
+    ///
+    /// When the `ExactMatch` policy is used, the start address MUST be aligned to the
+    /// alignment passed as a parameter.
+    ///
+    /// # Arguments:
+    /// - `size`: size to allocate.
+    /// - `align`: alignment to be used for the allocated resources.
```
Comment (lauralt) [inline src/address_allocator.rs:56]: nit: the argument name is `alignment`.

File: src/address_allocator.rs [hunk 2]
```diff
@@ -47,22 +47,22 @@ impl AddressAllocator {
...
+    ///            Valid alignments are a power of 2.
+    /// - `policy`: allocation policy.
```
Comment (lauralt) [inline src/address_allocator.rs:58]: Same: policy -> alloc_policy.

File: src/lib.rs [hunk 3]
```diff
@@ -17,6 +17,9 @@ use std::cmp::min;
 use std::result;
 use thiserror::Error;
 
+/// Default alignment that can be used for creating a `Constraint`.
+pub const DEFAULT_CONSTRAINT_ALIGN: u64 = 4;
```
Comment (lauralt) [inline src/lib.rs:21]: Do we also want to define a constructor that uses these defaults?

**Review by lauralt — APPROVED** [review]

File: src/lib.rs (hunk 3 above)
Comment (andreeaflorescu) [reply src/lib.rs:21]: I am not sure what do you mean by that. Do you mean to have a new constructor where we don't have to pass the default alignment?

Comment (lauralt) [reply src/lib.rs:21]: The default align and the default policy; I noticed `Constraint::new(0x1000, 0x1000, AllocPolicy::default())` and I was thinking if it makes sense to add a `Default` impl for Constraint to shorten things a bit there. Not sure if it makes sense, I don't have much context on vm-allocator.

Comment (andreeaflorescu) [reply src/lib.rs:21]: I find that to be useful mostly in tests (i.e. creating a constraint with default values). I created a variable for the align policy mostly as an example, but we can also delete it. The code looked a bit different in my first iteration and I needed that.

**Review by lauralt — APPROVED** [review]

**Review by AlexandruCihodaru — APPROVED** [review]
LGTM 🚢

**Pull request merged**